qa/tasks/vstart_runner.py

```diff
@@ -1179,6 +1179,38 @@ def clear_old_log():
         init_log()
         log.info('logging in a fresh file now...')
 
+# This will cleanup the stale netnses, which are from the
+# last failed test cases.
+def cleanup_stale_netnses_and_bridge():
```

Comment:
Why not make this part of `qa/tasks/cephfs/mount.py` so it cleans up stale netns in teuthology too?

Reply:
IMO, it makes no sense to do this for the teuthology, because at beginning the test nodes will be rebooted, right ?  And all the stale netnses will be cleaned up after nodes rebooted. 

And also once current test case fail in any reason, no new test cases should be rerun and continued in teuthology.

Reply:
> And also once current test case fail in any reason, no new test cases should be rerun and continued in teuthology.

That may not always be the case.

In any case, I think we need to fix this from the other direction: `vstart_runner.py` should not be leaving these network bridges behind when it exits. I'm not sure why, but it does not appear to do any unmounts before exiting after completing the test. Can you try changing that instead?

Reply:
> > And also once current test case fail in any reason, no new test cases should be rerun and continued in teuthology.
> 
> That may not always be the case.
> 
> In any case, I think we need to fix this from the other direction: `vstart_runner.py` should not be leaving these network bridges behind when it exits. I'm not sure why, but it does not appear to do any unmounts before exiting after completing the test. Can you try changing that instead?

Normally if the test case finish without any assertion or errors, it will remove all the netnses and bridge, etc it created.  The problem is that if the test case is aborted with some errors, the vstart_runner.py will not have any chance to do the clean up.

Reply:
The stale netnses and bridge are all left from the previous failing test cases.

Reply:
The mount.py is not a good place to fix this for teuthology IMO, the mount.py just for each mountpoint, for example if there have 2 mountpoints, the second one may remove the first one's netns and the global bridge.